sqlserver/datadog_checks/sqlserver/activity.py

```diff
@@ -361,6 +432,27 @@ def _create_activity_event(self, active_sessions, active_connections):
         }
         return event
 
+    def _create_deadlock_event(self, deadlock_xmls):
+        #TODO WHAT if deadlock xml is just too long ? 
+        #MAX_PAYLOAD_BYTES ? 
+        #TODO compression , couldnt reallly see it in Go code so far but may be its somewhere deeper. 
+        # may be 	flushAndSerializeInParallel FlushAndSerializeInParallel in aggregator.go ? 
+        # compression is Trade off CPU , network ?
+        event = {
+            "host": self._check.resolved_hostname,
+            "ddagentversion": datadog_agent.get_version(),
+            "ddsource": "sqlserver",
+            "dbm_type": "deadlocks",
+            "collection_interval": self._deadlocks_collection_interval,
+            "ddtags": self.tags,
+            "timestamp": time.time() * 1000,
+            'sqlserver_version': self._check.static_info_cache.get(STATIC_INFO_VERSION, ""),
+            'sqlserver_engine_edition': self._check.static_info_cache.get(STATIC_INFO_ENGINE_EDITION, ""),
+            "cloud_metadata": self._config.cloud_metadata,
+            "sqlserver_deadlocks": deadlock_xmls,
```

Comment:
do we plan on sending raw xml from the agent and do the parsing in backend?

Reply:
yes, except for obfuscation, no treatment

Reply:
@lu-zhengda what's your opinion?

Reply:
imo it's a good approach to send xml as raw string and parse at the backend.